tests/poll.rs

```diff
@@ -310,10 +311,69 @@ fn registry_operations_are_thread_safe() {
     handle3.join().unwrap();
 }
 
-// On kqueue platforms registering twice (not *re*registering) works.
+// This test checks the following reregister constraints:
+// The `reregister` arguments fully override the previous values. In other
+// words, if a socket is registered with `READABLE` interest and the call
+// to `reregister` specifies `WRITABLE`, then read interest is no longer
+// requested for the handle.
+#[test]
+fn reregister_overrides_interest() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let barrier = Arc::new(Barrier::new(2));
+    let (thread_handle, address) = start_listener(1, Some(barrier.clone()), false);
+
+    let mut stream = TcpStream::connect(address).unwrap();
```

Comment:
Why not use an `UdpSocket`, then we don't have to start a `TcpListener`.

Reply:
yup, good idea, dropped the listener